src/client.cpp

```diff
@@ -908,6 +905,56 @@ void CClient::Stop()
     SignalLevelMeter.Reset();
 }
 
+/// @method
+/// @brief Connects to strServerAddress
+/// @emit Connecting (strServerName) if the client wasn't running and SetServerAddr returned true.
+/// @param strServerAddress - the server address to connect to
+/// @param strServerName - the String argument to be passed to Connecting()
+/// @result true if client wasn't running and SetServerAddr returned true, false otherwise
+bool CClient::Connect ( QString strServerAddress, QString strServerName )
+{
+    if ( !IsRunning() )
+    {
+        // Set server address and connect if valid address was supplied
+        if ( SetServerAddr ( strServerAddress ) )
+        {
+
+            Start();
+
+            emit Connecting ( strServerName );
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/// @method
+/// @brief Disconnects client
+/// @emit Disconnected
+/// @result true if client wasn't running, false otherwise
+bool CClient::Disconnect()
+{
+    if ( IsRunning() )
```

Comment:
I have the feeling that we MUST clarrify this. 
The code in #3364 is different.

Reply:
I've no idea what this is trying to encapsulate.

Reply:
I mean we could probably emit in the Stop() method already...